src/index.js

```diff
@@ -176,17 +184,27 @@ const compare = (condition, value, attributes) => {
       );
 
     case 'startsWith':
+      if (compareValue === undefined) return undefined;
```

Comment:
looks like this is how it is handled in other cases, but I was curious why `return undefined` and not `false`?

Reply:
Seems like that's how it was designed to be: 

```
true if the comparision matches, false if there is a mismatch, and undefined if the target value is not known to compute the result
```